RFCs/0004-add-pdk-publish.md

```diff
@@ -0,0 +1,123 @@
+# Add `pdk publish` subcommand
+
+## Summary
+
+Add a new `pdk publish` subcommand that allows publishing a given module tarball to the Forge.  
+*Note that this RFC supersedes [RFC-0003](https://github.com/puppetlabs/pdk-planning/blob/master/RFCs/0003-add-pdk-release.md). While RFC-0003 may eventually still be implemented (likely as part of a PDK 2.x major release), this work represents a smaller scope that is more appropriate to the current PDK 1.x series.*
+
+## Background & Assumptions
+
+- With the addition of module management endpoints to the Forge API and authentication via API keys, [authenticating](https://forgeapi.puppet.com/#section/Authentication) and [publishing](https://forgeapi.puppet.com/#operation/addRelease) to the Forge has a public and stable interface.
+
+- Publishing to the Forge is a natural progression beyond the current `pdk build` subcommand, and is an action that may be performed manually or through an automated process.
+
+## Motivation & Goals
+
+- PDK should provide an easy way for module authors to package and distribute their modules via the Puppet Forge.
+
+- PDK should provide composable commands and actions so that module authors can automate the packaging and distribution of their modules, and integrate those steps into a comprehensive CI/CD pipeline.
+
+
+## Proposed Implementation Design
+
+Implement a new pdk subcommand, `pdk publish`, with the following syntax, options, and arguments.
+
+- Basic syntax:
+    - `pdk publish [--package-dir=<path>] [--force] [tarball.tar.gz]`
+
+
+- Options and arguments:
+    - `--package-dir=<path>`
+      - Use given path as path to find tarball to publish.
+
+    - `--force`
+      - Do not prompt for confirmation.
+
+    - `tarball.tar.gz`
+      - Relative or absolute path to tarball to publish.
+
+- Option combination behaviors:
+    - `(no options/arguments)`
+      - Looks in current module’s `metadata.json` for namespace, module name, and version and then attempts to publish `<namespace>-<module_name>-<version>.tar.gz` from default package path.
+      - If file matching the `<namespace>-<module_name>-<version>.tar.gz` pattern cannot be found in the default package path, exits with an error suggesting that user needs to use `pdk build` first.
+
+
+    - `<path to tarball>`
```

Comment:
Not sure what you mean here.  There's no `<path to tarball>` option above.  Perhaps an example command line usage would be better? (Also handy for acceptance tests :-) )

Reply:
Ah, yeah, that was changed after Jesse and I had determined the tarball should supersede the `--package-dir`, but it should be clearer. Do you think changing all of the `<path to tarball>` instances to just `tarball.tar.gz`, and referring to that argument as the "pathed tarball" is clear enough?

I will definitely add or edit the examples as well.